nipype/interfaces/base.py

```diff
@@ -559,62 +568,61 @@ def get_hashval(self, hash_method=None):
 
         dict_withhash = []
         dict_nofilename = []
-        for name, val in sorted(self.get().items()):
-            if isdefined(val):
-                trait = self.trait(name)
-                if has_metadata(trait.trait_type, "nohash", True):
-                    continue
-                hash_files = (not has_metadata(trait.trait_type, "hash_files",
-                                               False) and not
-                              has_metadata(trait.trait_type, "name_source"))
-                dict_nofilename.append((name,
-                                        self._get_sorteddict(val, hash_method=hash_method,
-                                                             hash_files=hash_files)))
-                dict_withhash.append((name,
-                                      self._get_sorteddict(val, True, hash_method=hash_method,
-                                                           hash_files=hash_files)))
+        for name, val in sorted(self.get_traitsfree().items()):
```

Comment:
the reason i used `get` in the previous version was to speed up the hashval computation. since `get_traitsfree`  requires a dictionary cleanup. if using get doesn't change assumptions here, it may be cleaner.